### ISSUE TITLE
There's no way to pass class labels from list in tf.keras.preprocessing.image_dataset_from_directory using this

### ISSUE THREAD

**Issue by AAKrupnov** [tensorflow/tensorflow#48901]
If there're no subfolders and labels to preprocessing.image_dataset_from_directory function are passed from external list this part of code will generate empty 'results' variable and further in line 88-98 this will lead to an error of not finding any files.
But if there's at least one subfolder the same code will lead to labels being inherited from subfolder name.
This means labels at all times behave as "inferred" and there's no option to pass labels from list in preprocessing.image_dataset_from_directory that uses this function. 

https://github.com/tensorflow/tensorflow/blob/85c8b2a817f95a3e979ecd1ed95bff1dc1335cff/tensorflow/python/keras/preprocessing/dataset_utils.py#L84

**Comment by sachinprasadhs** [comment]
tf.keras.preprocessing.image_dataset_from_directory will be deprecated in Tensorflow 2.9 version, use tf.keras.utils.image_dataset_from_directory instead.
Coming to your actual issue, I can see the changes made to include another subdir here https://github.com/keras-team/keras/blob/master/keras/utils/dataset_utils.py#L88.
Thanks!
